### PR TITLE
Feature/multiplechoice

### PR DESCRIPTION
**Pull request by Beaufort8**
add multiple choice functionality

### CODE REVIEW

**Comment by iionly** [comment]
I've started looking at the code of your PR. Firstly, I have to admit that I had misunderstood the feature. I had thought it was about allowing multiple questions within a single poll instead of allowing the multiple choices per single poll question. The feature as I had wrongly understood it at first would surely be more difficult to implement. Still, the multiple choices per questions are also a useful new feature and I would like to include it.

Though I would like to add it, I had not yet the time to fully investigate the new code of your PR. Unfortunately, I'm currently so busy at work that I have basically no time for anything else. I hope to be able to deal with it soon. But I might only be able to spend time on it on weekends.

A few things I've noticed already:

* it might make sense to add a new plugin setting to allow for the site admin to allow multiple choices in polls or don't allow them at all,
* currently the maximum possible number of choices a user would be able to give his/her vote would be 11 (I hope I understood that right). It seems like an arbitrary limit. As the number of choices is not restricted it might happen that more choices than 11 would be available to vote on. On the other hand, with less than 11 choices it would be strange to offer up to 11 choices to be voted for. Maybe best would be some dynamic adjustment of the number (by adding some JS code) up to the maximum number of choices currently available in the poll.
* with only 1 vote a user can give per poll the total number of votes given is easy to calculate (currently just the numbers per choice are summed up). With more than a single vote per user possible this isn't as easy anymore. I would still see a single user having participated in the poll count as 1 vote even on multiple choice polls. Therefore, it will be necessary to change the code calculating the total number of votes given.

File: views/default/forms/poll/edit.php [hunk 1]
```diff
@@ -67,6 +67,25 @@
 ?>
 
 <div>
+<?php
+
+$options_values = array();
+for ($i = 1; $i < 11; $i++) {
```
Comment (juho-jaakkola) [inline views/default/forms/poll/edit.php]: I agree with iionly that 11 is a bit arbitrary limit. I'd suggest adding a text field that allows poll owner to enter a number. The saving action can then check the given limit and show an error if the limit is equal or greater than the amount of entered options.

**Comment by juho-jaakkola** [comment]
I reworked this a bit. See #32

**Comment by Beaufort8** [comment]
Input is now a text-field and maximum will be checked on save

**Comment by iionly** [comment]
I'll have to test out this PR and the rework of @juho-jaakkola. But I don't think I will be able to do so before next weekend.

**Comment by iionly** [comment]
Support added by https://github.com/iionly/poll/commit/6ba6385ba00a447c8de6c6bc1c294ce6b73fe58d.

@Beaufort8 
I've reworked Juho's PR and also added a plugin setting to enable/disable multi-answer polls. If you would have a bit of time to spare, I would be happy if you could test out the poll plugin with the recents commits made.

**Pull request closed without merging**